### PR TITLE
InfluxDB: Add delete buttons to measurement, part and tag in query editor

### PR DESCRIPTION
**Pull request by AlexGodbehere**
**What is this feature?**

This PR implements a delete button to the `measurement`, `part` and `tag` sections in the InfluxDB datasource editor. This improves user experience by allowing users to delete measurements, parts and tags directly from the user interface. The commit also slightly refactors the options generation for the `where` clause, removing the `-- remove filter --` option as it becomes redundant with the new delete button.

Before:
image

After:
image

**Why do we need this feature?**

Before this change there was no way to remove `GROUP BY` entries or clear the currently selected measurement, which caused an issue because available tag values were filtered by the current measurement. The only solution was to remove the query and start again or splice out the offending code in the raw query editor.

**Who is this feature for?**

Users of the InfluxDB datasource that use the InfuxQL graphical query editor.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by catherineymgui** [comment]
Thanks for your contribution!

It would be great if we could match some of the other data sources (Prometheus is the one we try to aspire to). Would you be able to use the "X" instead of the trash icon?

Screenshot 2023-08-04 at 08 50 14

**Comment by AlexGodbehere** [comment]
Sure thing. I've moved over to `AccessoryButton` as Prometheus uses. It doesn't look as swish as Prometheus but it's a step closer.

image

**Comment by AlexGodbehere** [comment]
@catherineymgui are you a reviwer for this? What's the process for getting this approved?

File: public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/FromSection.tsx [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import React from 'react';
 
+import { AccessoryButton } from '@grafana/experimental';
+
 import { DEFAULT_POLICY } from '../../../../../types';
+
```
Comment (itsmylife) [inline public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/FromSection.tsx:6]:
```suggestion
```

Comment (itsmylife) [reply public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/FromSection.tsx:6]: Just to make `eslint` happy

File: public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/PartListSection.tsx [hunk 2]
```diff
@@ -125,17 +126,28 @@ export const PartListSection = ({
...
+          <AccessoryButton
+            style={{ marginRight: '4px' }}
+            aria-label="remove"
+            icon="times"
+            variant="secondary"
+            onClick={() => {
+              onRemovePart(index);
+            }}
+          />
+        </>
```
Comment (itsmylife) [inline public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/PartListSection.tsx:150]:
```suggestion
        <React.Fragment key={index}>
          <Part
            name={part.name}
            params={part.params}
            onRemove={() => {
              onRemovePart(index);
            }}
            onChange={(pars) => {
              onChange(index, pars);
            }}
          />
          <AccessoryButton
            style={{ marginRight: '4px' }}
            aria-label="remove"
            icon="times"
            variant="secondary"
            onClick={() => {
              onRemovePart(index);
            }}
          />
        </React.Fragment>
```

Comment (itsmylife) [reply public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/PartListSection.tsx:150]: Moving the `key` to `Fragment`

**Review by itsmylife — APPROVED** [review]
LGTM 🚀 
Thanks for the contribution!

**Pull request merged**